app_integrations/apps/onelogin.py

```diff
@@ -0,0 +1,192 @@
+                    'assuming_acting_user_id': null,
+                    'role_id': <int:role_id>,
+                    'app_id': <int:app_id>,
+                    'group_id': <int:group_id>,
+                    'otp_device_id': <int:otp_device_id>,
+                    'policy_id': <int:policy_id>,
+                    'actor_system': <str:actor_system>,
+                    'custom_message': <str:custom_message>,
+                    'role_name': <str:role_name>,
+                    'app_name': <str:app_name>,
+                    'group_name': <str:group_name>,
+                    'actor_user_name': <str:actor_user_name>,
+                    'user_name': <str:user_name>,
+                    'policy_name': <str:policy_name>,
+                    'otp_device_name': <str:otp_device_name>,
+                    'operation_name': <str:operation_name>,
+                    'directory_sync_run_id': <int:directory_sync_run_id>,
+                    'directory_id': <int:directory_id>,
+                    'resolution': <str:resolution>,
+                    'client_id': <int:client_id>,
+                    'resource_type_id': <int:resource_type_id>,
+                    'error_description': <str:error_description>
+                }
+            ]
+        """
+        # OneLogin API expects the ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ
+        formatted_date = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
+        params = {'since': formatted_date}
+
+        # Make sure we have authentication headers
+        if not self._auth_headers:
+            return False
+
+        events = self._get_onelogin_paginated_events(params)
+
+        while self._more_to_poll and self._next_page_url:
+            LOGGER.debug('More events to retrieve for \'%s\': %s', self.type(), self._more_to_poll)
+            pagination = self._get_onelogin_paginated_events(None)
+
+            # Add the events to our results, this is equivalent to using events.extend()
+            events += pagination
+
+        # Return the list of logs to the caller so they can be send to the batcher
+        return events
+
+    def _get_onelogin_paginated_events(self, params):
+        """Get events from the API pagination url
+
+        Returns:
+            The same as the method _get_onelogin_events()
+        """
+        response = requests.get(self._next_page_url, headers=self._auth_headers, params=params)
+        if not self._check_http_response(response):
+            return False
+
+        # Extract events from response
+        events = response.json()['data']
+
+        # Do we have a pagination link to follow?
+        self._more_to_poll = len(events) >= self._MAX_EVENTS_LIMIT
+
+        # If we are already retrieved all the events, then set the value to prevent more requests
+        self._next_page_url = response.json()['pagination']['next_link']
+
+        return events
+
+    def required_auth_info(self):
+        return {
```

Comment:
As we spoke about offline, please add the localization (us/eu) to these so they can be user-provided.